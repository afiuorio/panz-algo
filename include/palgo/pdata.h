#ifndef PDATA_H_
#define PDATA_H_

void **alloc_pMatrix(size_t numRow, size_t numCol, size_t elem_size);
void free_pMatrix(void **matrix);

#endif
