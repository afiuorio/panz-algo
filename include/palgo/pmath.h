#ifndef PMATH_H_
#define PMATH_H_

int p_abs(int n);

double p_cos(double x);
double p_sin(double x);

double p_sqrt(double n);

double p_pow(double x, double n);

#endif
