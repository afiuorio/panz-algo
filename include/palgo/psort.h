#ifndef PSORT_H_
#define PSORT_H_

void p_sort(void *base, size_t n, size_t size, int(*cmp)(const void *,const void *));

void p_isort(void *base, size_t n, size_t size,int(*cmp)(const void*,const void*));

int p_rsort(unsigned int *array, size_t n);

#endif
