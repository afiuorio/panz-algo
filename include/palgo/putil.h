#ifndef PUTIL_H_
#define PUTIL_H_

#define PRAND_MAX 0x7fff

void p_srand(unsigned int seed);
int p_rand();

void *p_search(void *key,void *array, size_t num ,size_t size,int(*cmp)(const void *,const void *) );

char *p_strcpy(char *s, const char *ct);
char *p_strcat(char *s, const char *ct);
int p_strcmp(const char *cs, const char *ct);

#endif
