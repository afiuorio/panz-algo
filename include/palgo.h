#ifndef PALGO_H_
#define PALGO_H_

#include "palgo/pdata.h"
#include "palgo/pmath.h"
#include "palgo/psort.h"
#include "palgo/putil.h"

#endif
