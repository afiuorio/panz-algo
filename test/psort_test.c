#include<stdio.h>
#include<palgo.h>

#define PRINT_ARRAY do{ \
    int i = 0; \
	for(i=0;i< 14; i++) \
		printf("%d,", array[i]); \
    printf("\n"); \
} while(0)

int cmpfunc (const void * a, const void * b)
{
   return ( *((int*)a) - *((int*)b) );
}

void test_quicksort(){
    int array[14] = { 2, 453,443, 665, 433, 322, 11, 34, 245, 243, 664, 34564,213245, 1002};
	p_sort(array, 14, sizeof(int), cmpfunc);
	PRINT_ARRAY;
}

void test_insertionSort(){
    int array[14] = { 2, 453,443, 665, 433, 322, 11, 34, 245, 243, 664, 34564,213245, 1002};
	p_isort(array, 14, sizeof(int), cmpfunc);
	PRINT_ARRAY;
}

void test_radixSort(){
    unsigned int array[14] = { 2, 453,443, 665, 433, 322, 11, 34, 245, 243, 664, 34564,213245, 1002};
	p_rsort(array, 14);
	PRINT_ARRAY;
}

int main(){
    test_quicksort();
    test_radixSort();
    test_insertionSort();
}
