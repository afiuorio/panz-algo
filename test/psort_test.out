2,11,34,243,245,322,433,443,453,664,665,1002,34564,213245,
2,11,34,243,245,322,433,443,453,664,665,1002,34564,213245,
2,11,34,243,245,322,433,443,453,664,665,1002,34564,213245,
